RegistrySubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java

```diff
@@ -1,145 +1,295 @@
+			+ "\r\n  } ],\r\n  \"@id\" : \"urn:ngsi-ld:Subscription:173223\","
+			+ "\r\n  \"https://uri.etsi.org/ngsi-ld/notification\" : [ "
+			+ "{\r\n    \"https://uri.etsi.org/ngsi-ld/endpoint\" : [ "
+			+ "{\r\n      \"https://uri.etsi.org/ngsi-ld/accept\" : [ "
+			+ "{\r\n        \"@value\" : \"application/json\""
+			+ "\r\n      } ],\r\n      \"https://uri.etsi.org/ngsi-ld/uri\" : [ {"
+			+ "\r\n        \"@value\" : \"http://localhost:8080/acc\"\r\n      } ]"
+			+ "\r\n    } ]\r\n  } ],\r\n  \"@type\" : [ \"https://uri.etsi.org/ngsi-ld/Subscription\" ]" + "\r\n}";
+
+	String payload1 = "{\r\n  \"https://uri.etsi.org/ngsi-ld/entities\" : [ "
+			+ "{\r\n    \"@type\" : [ \"https://uri.etsi.org/ngsi-ld/default-context/Vehicle\" ]"
+			+ "\r\n  } ],\r\n  \"@id\" : \"urn:ngsi-ld:Subscription:173224\","
+			+ "\r\n  \"https://uri.etsi.org/ngsi-ld/notification\" : [ "
+			+ "{\r\n    \"https://uri.etsi.org/ngsi-ld/endpoint\" : [ "
+			+ "{\r\n      \"https://uri.etsi.org/ngsi-ld/accept\" : [ "
+			+ "{\r\n        \"@value\" : \"application/json\""
+			+ "\r\n      } ],\r\n      \"https://uri.etsi.org/ngsi-ld/uri\" : [ {"
+			+ "\r\n        \"@value\" : \"http://localhost:8080/acc\"\r\n      } ]"
+			+ "\r\n    } ]\r\n  } ],\r\n  \"@type\" : [ \"https://uri.etsi.org/ngsi-ld/Subscription\" ]" + "\r\n}";
+
+	/**
+	 * this method is used to test create subscription test
+	 */
+	@Test
+	@Order(1)
+	public void serviceTest() throws ResponseException {
+
+		List<Object> context = new ArrayList<>();
+		Context context1 = new Context();
+		Subscription subscription = null;
+		ArrayListMultimap<String, String> subIds = ArrayListMultimap.create();
+		multimaparr.put("content-type", "application/json");
+		Gson gson = new Gson();
+		Map<String, Object> resolved = gson.fromJson(payload, Map.class);
+		Subscription s = Subscription.expandSubscription(resolved, context1, true);
+		subscription = new Subscription();
+		
+		SubscriptionRequest subRequest = new SubscriptionRequest(s, context, subIds, 0);
+		subscriptionService.subscribeInternal(subRequest).await().indefinitely();
+	}
+
+	
+	
+	/**
+	 * this method is used to test create subscription test for deconstructor
+	 */
+	
+	@Test
+	@Order(2)
+	public void serviceTestforDeconstructor() throws ResponseException {
+
+		List<Object> context = new ArrayList<>();
+		Context context1 = new Context();
+		Subscription subscription = null;
+		ArrayListMultimap<String, String> subIds = ArrayListMultimap.create();
+		multimaparr.put("content-type", "application/json");
+		Gson gson = new Gson();
+		
+		Map<String, Object> resolved = gson.fromJson(payload1, Map.class);
+		Subscription s = Subscription.expandSubscription(resolved, context1, true);
+		subscription = new Subscription();
+		SubscriptionRequest subRequest = new SubscriptionRequest(s, context, subIds, 0);
+		
+		subscriptionService.subscribeInternal(subRequest).await().indefinitely();
```

Comment:
Please add verify condition and check expected and actual result in all test cases.

Reply:
Done